### PR TITLE
Iterate on API sketch

### PR DESCRIPTION
**Pull request by pmorie**
Refine a few aspects of the API sketch started this afternoon.

@arschles @slack @duglin @MHBauer @vaikas-google  @pwittrock @bmelville

### CODE REVIEW

**Comment by duglin** [comment]
Before we go much further I'd like to understand how we can have the service controller be generic. Pulling in K8s imports doesn't feel generic :-)

File: pkg/api/types.go [hunk 1]
```diff
@@ -27,87 +52,152 @@ type BrokerStatus struct {
...
+	CFPlanID        string
+	CFType          string
+	CFSpaceGUID     string
+	CFLastOperation string
 }
 
-type InstanceStatus struct {
-	Status string // TODO: make this an "enum" (constant + type)
+type InstanceState struct {
+	Status InstanceState
```
Comment (arschles) [inline pkg/api/types.go]: this type should be called `InstanceStatus`. right now, it's a circular dependency

**Comment by arschles** [comment]
LGTM assuming we'll continue iterating after this

**Comment by pmorie** [comment]
@arschles @duglin 

I would prefer strongly to avoid thinking about this for now to keep the water clear.  I think there are a number of elegant options that can eliminate the surfacing of k8s concepts to users of other platforms, but I would prefer to discuss that in inning 5 or 6 of the v1 series, and we are in inning 2 now.

> Also, @duglin do you believe it would be good enough to replace the k8s-imported data types with our own copies? for example, instead of using kapi.ObjectMeta, we copy the relevant bits into our own type and use it herein

**Comment by arschles** [comment]
all sounds fine to me :+1:

**Pull request merged**